lib/utilities/will-interrupt-process.js

```diff
@@ -0,0 +1,140 @@
+'use strict';
+
+// work around misbehaving libraries, so we can correctly cleanup before actually exiting.
```

Comment:
I wouldn't put this comment here with the code as-is.  It implies that we already capture exit by requiring, but we don't until we call `captureExit`.

That said I think there's a bigger issue mentioned above, which is that we want `captureExit` to be called much earlier (pretty much ASAP).